### PR TITLE
Bug/arrows to right sometimes enter into box

### PR DESCRIPTION
**Pull request by kentis**
* does not update boxes when creating arrows
* editor support for breakpoints

### CODE REVIEW

File: src/components/Draggable/Draggable.module.scss [hunk 1]
```diff
@@ -26,6 +26,11 @@
   }
 }
 
+// .arrow_mode {
+//   border: 1rem;
+//   border-style: groove;
+// }
+
```
Comment (boyum) [inline src/components/Draggable/Draggable.module.scss:33]:
```suggestion
```

**Comment by boyum** [comment]
I'm able to create arrow indicators on all cells now, even those without boxes in them.

This otherwise works great when the grid is of a certain width, however if the grid is 1000px wide, it never works when creating an arrow in positive x or positive y direction.

**Comment by kentis** [comment]
> I'm able to create arrow indicators on all cells now, even those without boxes in them.

This is actually by design in order to support setting breakpoints when creating arrows.

File: src/components/Arrow/Arrow.tsx [hunk 2]
```diff
@@ -39,7 +40,9 @@ export const Arrow: React.FC<ArrowProps> = ({
   setSelectedItemId,
   updateArrowType,
   gridWidth,
+  arrowStartId,
 }) => {
+  console.info("arrowStartId", arrowStartId);
```
Comment (boyum) [inline src/components/Arrow/Arrow.tsx:45]:
```suggestion
```

File: src/components/Grid/Grid.tsx [hunk 3]
```diff
@@ -225,19 +229,19 @@ export const Grid: FC<GridProps> = ({
           const cellsOfItem = occupiedCells.filter(
             c => c.occupiedById === elementId,
           );
+          // const cellOfItem = findCellsElementOccupies(elementId, gridW)
```
Comment (boyum) [inline src/components/Grid/Grid.tsx:232]:
```suggestion

```

File: src/utils/arrow.utils.ts [hunk 4]
```diff
@@ -110,19 +110,23 @@ export const yAdjustmentStart = (
...
+  const isHorizontal = calculateIsHorizontal(
+    sourcePosition,
+    item.endGridPosition,
+  );
   if (
     item.arrowType === ArrowType.Directional ||
     item.arrowType === ArrowType.BiDirectional
   ) {
-    if (isHorizontal && item.startGridPosition.x <= item.endGridPosition.x) {
+    if (isHorizontal && sourcePosition.x <= item.endGridPosition.x) {
```
Comment (boyum) [inline src/utils/arrow.utils.ts:129]:
```suggestion
    const goesToTheRight = isHorizontal && sourcePosition.x <= item.endGridPosition.x;
    if (goesToTheRight) {
```

File: src/utils/arrow.utils.ts [hunk 5]
```diff
@@ -134,7 +138,7 @@ export const xAdjustmentEnd = (
   }
 
   if (item.arrowType === ArrowType.NonDirectional) {
-    if (isHorizontal && item.startGridPosition.x <= item.endGridPosition.x) {
+    if (isHorizontal && sourcePosition.x <= item.endGridPosition.x) {
```
Comment (boyum) [inline src/utils/arrow.utils.ts:141]:
```suggestion
    const goesToTheRight = isHorizontal && sourcePosition.x <= item.endGridPosition.x;
    if (goesToTheRight) {
```

**Review by boyum — APPROVED** [review]
🤩🤩🤩🤩🤩🤩🤩🤩🤩🤩🤩🤩

**Review by boyum — APPROVED** [review]

**Pull request merged**